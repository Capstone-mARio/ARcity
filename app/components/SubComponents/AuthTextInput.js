import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { View } from 'react-native';

import {
  FormLabel,
  FormInput,
  FormValidationMessage,
} from 'react-native-elements';
import { isEmpty } from '../utils/validate';
import styles from './AuthTextInputStyles';
import { iOSUIKit, material } from 'react-native-typography';

class AuthTextInput extends Component {
  render() {
    const {
      showLabel,
      placeholder,
      autoFocus,
      onChangeText,
      secureTextEntry,
      placeholderTextColor,
      keyboardType,
    } = this.props;

    return (
      <View style={styles.container}>
        {showLabel && <FormLabel>{this.props.label}</FormLabel>}
        <FormInput
          autoCapitalize="none"
          clearButtonMode="while-editing"
          underlineColorAndroid={'#fff'}
          placeholder={placeholder}
          autoFocus={autoFocus}
          onChangeText={onChangeText}
          secureTextEntry={secureTextEntry}
          containerStyle={styles.inputContainer}
          inputStyle={material.titleWhite}
          placeholderTextColor={'grey'}
          keyboardType={keyboardType}
          value={this.props.value}
        />
        {!isEmpty(this.props.error) && (
          <FormValidationMessage>{this.props.error}</FormValidationMessage>
        )}
      </View>
    );
  }
}

AuthTextInput.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  placeholder: PropTypes.string,
  autoFocus: PropTypes.bool,
  onChangeText: PropTypes.func.isRequired,
  secureTextEntry: PropTypes.bool,
  value: PropTypes.string,
  error: PropTypes.string,
};

AuthTextInput.defaultProps = {
  autoFocus: false,
  secureTextEntry: false,
  placeholderTextColor: 'grey',
  keyboardType: 'default',
};

export default AuthTextInput;
