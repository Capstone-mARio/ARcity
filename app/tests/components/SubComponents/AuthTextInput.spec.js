import React from 'react';
import AuthTextInput from '../../../components/SubComponents/AuthTextInput';
import renderer from 'react-test-renderer';

const fields = [
  {
    key: 'username',
    label: 'Username',
    placeholder: 'Username',
    autoFocus: false,
    secureTextEntry: false,
    value: '',
    type: 'text',
  },
];

const firstField = fields[0];

test('renders AuthTextInput correctly', () => {
  const tree = renderer
    .create(
      <AuthTextInput
        key={firstField.key}
        label={firstField.label}
        showLabel={true}
        placeholder={firstField.placeholder}
        autoFocus={firstField.autoFocus}
        onChangeText={text => this.onChange(firstField.key, text)}
        secureTextEntry={firstField.secureTextEntry}
        keyboardType={undefined}
        value={'Test Username'}
        error={'Test Error'}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
