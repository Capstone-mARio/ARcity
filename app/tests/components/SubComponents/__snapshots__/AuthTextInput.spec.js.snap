// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthTextInput correctly 1`] = `
<View
  style={
    Object {
      "margin": 3,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Username
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#A5A7A9",
        "borderBottomWidth": 1,
        "height": 70.71428571428572,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 710,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoFocus={false}
      clearButtonMode="while-editing"
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="grey"
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(255,255,255,1)",
            "fontFamily": "System",
            "fontSize": 20,
            "fontWeight": "600",
            "letterSpacing": 0.361328,
            "lineHeight": 28,
          },
        ]
      }
      textAlign="auto"
      underlineColorAndroid="#fff"
      value="Test Username"
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Test Error
    </Text>
  </View>
</View>
`;
